src/rebar_compiler_erl.erl

```diff
@@ -143,7 +143,7 @@ compile(Source, [{_, OutDir}], Config, ErlOpts) ->
 
 compile_and_track(Source, [{Ext, OutDir}], Config, ErlOpts) ->
     rebar_compiler_epp:flush(),
-    BuildOpts = [{outdir, OutDir} | ErlOpts],
+    BuildOpts = [{outdir, OutDir}, no_spawn_compiler_process | ErlOpts],
```

Comment:
Unless we add this option to `effects_code_generation/1` and possibly sync it well `needed_files`, I think we could risk having a few extra rebuilds because of this.

Reply:
It may also worth checking OTP 19 (I recall it wasn't liking the unknown flag).

Reply:
We currently have it in CI and it seems to pass fine on that dimension at least.

Reply:
I added it to `effects_code_generation` to return `false`, but I don't know what you mean with `needed_files`, @ferd

Reply:
yeah it's alright, `effects_code_generation` is used by `needed_files` so it should all be alright.

Reply:
ah I merged and it _does_ trigger recompiles. Let me patch that up